pkg/grafana/datasource-handler.go

```diff
@@ -55,32 +56,32 @@ func (h *DatasourceHandler) newDatasourceResource(path, uid, filename string, so
 	return resource
 }
 
-// Parse parses an interface{} object into a struct for this resource type
-func (h *DatasourceHandler) Parse(path string, i interface{}) (grizzly.ResourceList, error) {
+// Parse parses a manifest object into a struct for this resource type
+func (h *DatasourceHandler) Parse(m manifest.Manifest) (grizzly.ResourceList, error) {
 	resources := grizzly.ResourceList{}
-	msi := i.(map[string]interface{})
-	for k, v := range msi {
-		source := Datasource{}
-		source["basicAuth"] = false
-		source["basicAuthPassword"] = ""
-		source["basicAuthUser"] = ""
-		source["database"] = ""
-		source["orgId"] = 1
-		source["password"] = ""
-		source["secureJsonFields"] = map[string]interface{}{}
-		source["typeLogoUrl"] = ""
-		source["user"] = ""
-		source["withCredentials"] = false
-		source["readOnly"] = false
-
-		err := mapstructure.Decode(v, &source)
-		if err != nil {
-			return nil, err
-		}
-		resource := h.newDatasourceResource(path, source.UID(), k, source)
-		key := resource.Key()
-		resources[key] = resource
+	spec := m["spec"].(map[string]interface{})
+	source := Datasource{}
+	source["basicAuth"] = false
+	source["basicAuthPassword"] = ""
+	source["basicAuthUser"] = ""
+	source["database"] = ""
+	source["orgId"] = 1
+	source["password"] = ""
+	source["secureJsonFields"] = map[string]interface{}{}
+	source["typeLogoUrl"] = ""
+	source["user"] = ""
+	source["withCredentials"] = false
+	source["readOnly"] = false
```

Comment:
Wouldn't this read cleaner like so:
```
source := Datasource{
  basicAuth: false,
  ...
}
```

Reply:
I keep fixing that one, it never seems to make it in. This is a move rather than a rewrite, but yes, I can fix that.